internal/config/loader.go

```diff
@@ -10,38 +11,107 @@ package config
 import (
 	"errors"
 	"fmt"
+	"github.com/edgexfoundry/go-mod-registry"
+	"github.com/edgexfoundry/go-mod-registry/pkg/factory"
 	"io/ioutil"
-	"net/http"
 	"os"
+	"os/signal"
 	"strconv"
+	"syscall"
 	"time"
 
-	"github.com/BurntSushi/toml"
 	"github.com/edgexfoundry/device-sdk-go/internal/common"
-	"github.com/edgexfoundry/device-sdk-go/internal/registry"
-)
+	"github.com/pelletier/go-toml"
 
-const consulStatusPath = "/v1/agent/self"
+)
 
 var (
-	// Need to set timeout because it hang until server close connection
-	// https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779
-	netClient      = &http.Client{Timeout: time.Second * 10}
 	RegistryClient registry.Client
 )
 
 // LoadConfig loads the local configuration file based upon the
 // specified parameters and returns a pointer to the global Config
 // struct which holds all of the local configuration settings for
 // the DS. The bool useRegisty indicates whether the registry
-// should be used to read initial config settings. This also controls
+// should be used to read config settings. This also controls
 // whether the service registers itself the registry. The profile and confDir
 // are used to locate the local TOML config file.
-func LoadConfig(useRegistry bool, profile string, confDir string) (config *common.Config, err error) {
+func LoadConfig(useRegistry bool, profile string, confDir string) (*common.Config, error) {
 	fmt.Fprintf(os.Stdout, "Init: useRegistry: %v profile: %s confDir: %s\n",
 		useRegistry, profile, confDir)
-	confName := "configuration.toml"
 
+	configuration, err := loadConfigFromFile(profile, confDir)
+
+	// TODO: Verify this is correct.
+	stem := common.ConfigRegistryStem + common.ServiceName + "/"
+
+	var registryMsg string
+	if useRegistry {
+		registryMsg = "Register in registry..."
+		registryConfig := registry.Config{
+			Host:          configuration.Registry.Host,
+			Port:          configuration.Registry.Port,
+			Type:          configuration.Registry.Type,
+			Stem:          stem,
+			CheckInterval: configuration.Registry.CheckInterval,
+			CheckRoute:    common.APIPingRoute,
+			ServiceKey:    common.ServiceName,
+			ServiceHost:   configuration.Service.Host,
+			ServicePort:   configuration.Service.Port,
+		}
+
+		RegistryClient, err =  factory.NewRegistryClient(registryConfig)
+		if err != nil {
+			return nil, fmt.Errorf("connection to Registry could not be made: %v", err.Error())
+		}
+
+		// Check if registry service is running
+		if err := checkRegistryUp(configuration); err != nil {
+			return nil, err
+		}
+
+		// Register the service with Registry for discovery and health checks
+		err = RegistryClient.Register()
+		if err != nil {
+			return nil, fmt.Errorf("could not register service with Registry: %v", err.Error())
+		}
+
+		hasConfiguration, err := RegistryClient.HasConfiguration()
+		err = RegistryClient.Register()
```

Comment:
Is line 80 duplicated to line 74?

Reply:
Yes. Good catch. A little too mush copy and paste... ;-)